### PR TITLE
[dockerode] Add missing types for docker.run

### PR DESCRIPTION
**Pull request by MasterOdin**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: see inline comments
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/dockerode/index.d.ts [hunk 1]
```diff
@@ -1129,6 +1129,7 @@ declare namespace Dockerode {
 
     interface ContainerCreateOptions {
         name?: string | undefined;
+        platform?: string | undefined;
```
Comment (MasterOdin) [inline types/dockerode/index.d.ts:1132]: Per the latest docker API docs (https://docs.docker.com/engine/api/v1.42/#tag/Container/operation/ContainerCreate), this is an allowed query parameter, which dockerode supports (see https://github.com/apocas/dockerode/issues/721#issuecomment-1520328035)

File: types/dockerode/index.d.ts [hunk 2]
```diff
@@ -2018,36 +2023,29 @@ declare class Dockerode {
...
+        createOptions: Dockerode.ContainerCreateOptions,
         callback: Callback<any>,
     ): events.EventEmitter;
     run(
         image: string,
         cmd: string[],
         outputStream: NodeJS.WritableStream | NodeJS.WritableStream[],
-        createOptions: {},
         callback: Callback<any>,
     ): events.EventEmitter;
```
Comment (MasterOdin) [inline types/dockerode/index.d.ts:2042]: For the above, I:
1. removed the function definition that included `startOptions` and not `createOptions`
2. Reordered the functions so it was in order of most specified to least specified

For (1), [looking at the code](https://github.com/apocas/dockerode/blob/bb88194ff250c4f5b9a0eee2688c5dcc5f623ef9/lib/docker.js#L1487-L1494) it doesn't look to differentiate are you passing `createOptions` or `startOptions` if you only pass one object, rather it just assumes its `createOptions`. By adding actual types here, having both lead to a TS error in my example.

File: types/dockerode/index.d.ts [hunk 3]
```diff
@@ -1160,6 +1161,10 @@ declare namespace Dockerode {
         abortSignal?: AbortSignal;
     }
 
+    interface ContainerStartOptions {
+        detachKeys?: string;
+    }
```
Comment (MasterOdin) [inline types/dockerode/index.d.ts:1166]: See https://docs.docker.com/engine/api/v1.42/#tag/Container/operation/ContainerStart

**Review by sandersn — APPROVED** [review]
Since none of the current owners reviewed this, I'd recommend adding yourself as an owner, either here or in another PR.

**Comment by MasterOdin** [comment]
Ready to merge

**Pull request merged**